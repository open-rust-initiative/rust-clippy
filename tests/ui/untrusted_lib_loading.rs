#![allow(unused)]
#![warn(clippy::untrusted_lib_loading)]
#![feature(rustc_private)]

extern crate libloading;
use std::io::Read;

fn main() {
    let mut f = std::fs::File::open("tests/nagisa64.txt").unwrap();
    let mut name = String::new();
    f.read_to_string(&mut name).unwrap();

    unsafe {
        let lib = libloading::Library::new(&name);
        if lib.is_err() {
            println!("can't open {}", name);
        }
    }
}
