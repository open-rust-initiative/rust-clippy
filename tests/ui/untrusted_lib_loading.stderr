error: can't read outer files when loading the dynamic libraries!
  --> $DIR/untrusted_lib_loading.rs:9:17
   |
LL |     let mut f = std::fs::File::open("tests/nagisa64.txt").unwrap();
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: loading dynamic library here
  --> $DIR/untrusted_lib_loading.rs:14:19
   |
LL |         let lib = libloading::Library::new(&name);
   |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to previous error

